### PR TITLE
core/consensus: improve round change debug logging

### PR DESCRIPTION
**Pull request by corverroos**
Improve round change debug logging.

category: refactor
ticket: none

### CODE REVIEW

**Comment by corverroos** [comment]
Example output of a unit test:
```
10:37:54.417 DEBG qbft       QBFT consensus instance starting         {"duty": "1/attester", "peers": "[0:bright-boy 1:adorable-water 2:thoughtful-road 3:jolly-farmer]", "node": 3, "peer": "jolly-farmer"}
10:37:54.417 DEBG qbft       QBFT consensus instance starting         {"duty": "1/attester", "peers": "[0:bright-boy 1:adorable-water 2:thoughtful-road 3:jolly-farmer]", "node": 2, "peer": "thoughtful-road"}
10:37:54.417 DEBG qbft       QBFT consensus instance starting         {"duty": "1/attester", "peers": "[0:bright-boy 1:adorable-water 2:thoughtful-road 3:jolly-farmer]", "node": 0, "peer": "bright-boy"}
10:37:54.417 DEBG qbft       QBFT consensus instance starting         {"duty": "1/attester", "peers": "[0:bright-boy 1:adorable-water 2:thoughtful-road 3:jolly-farmer]", "node": 1, "peer": "adorable-water"}
10:37:54.418 DEBG qbft       QBFT upon rule triggered                 {"rule": "justified_pre_prepare", "round": 1, "node": 0, "peer": "bright-boy"}
10:37:54.421 DEBG qbft       QBFT upon rule triggered                 {"rule": "justified_pre_prepare", "round": 1, "node": 1, "peer": "adorable-water"}
10:37:54.422 DEBG qbft       QBFT upon rule triggered                 {"rule": "justified_pre_prepare", "round": 1, "node": 2, "peer": "thoughtful-road"}
10:37:54.422 DEBG qbft       QBFT upon rule triggered                 {"rule": "justified_pre_prepare", "round": 1, "node": 3, "peer": "jolly-farmer"}
10:37:54.423 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_prepares", "round": 1, "node": 0, "peer": "bright-boy"}
10:37:54.426 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_prepares", "round": 1, "node": 3, "peer": "jolly-farmer"}
10:37:54.426 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_prepares", "round": 1, "node": 2, "peer": "thoughtful-road"}
10:37:54.427 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_commits", "round": 1, "node": 2, "peer": "thoughtful-road"}
10:37:54.427 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_commits", "round": 1, "node": 0, "peer": "bright-boy"}
10:37:54.427 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_prepares", "round": 1, "node": 1, "peer": "adorable-water"}
10:37:54.427 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_commits", "round": 1, "node": 3, "peer": "jolly-farmer"}
10:37:54.427 DEBG qbft       QBFT upon rule triggered                 {"rule": "quorum_commits", "round": 1, "node": 1, "peer": "adorable-water"}

```

File: core/consensus/component.go [hunk 1]
```diff
@@ -106,20 +111,36 @@ func New(tcpNode host.Host, sender *p2p.Sender, peers []p2p.Peer, p2pKey *ecdsa.
...
+
+			steps := groupRoundMessages(msgs, len(peers), round, int(leader(duty, round, peers)))
+			for _, step := range steps {
+				fields = append(fields, z.Str(step.Type.String(), fmtStepPeers(step)))
+			}
+			if uponRule == qbft.UponRoundTimeout {
+				fields = append(fields, z.Str("timeout_reason", timeoutReason(steps, round, quorom)))
+			}
+
+			log.Debug(ctx, "QBFT round changed", fields...)
```
Comment (dB2510) [inline core/consensus/component.go:137]: add duty field in this log, it will help in debugging if there are mulitple QBFT instances running

Comment (corverroos) [reply core/consensus/component.go:137]: ctx already contains a duty in production so it would be there in prod

Comment (corverroos) [reply core/consensus/component.go:137]: in general you never have to add duty to the logs, since it should already be there

**Review by dB2510 — APPROVED** [review]

**Pull request merged**